### ISSUE TITLE
Accept an option to dump log output to github Job Summaries

### ISSUE THREAD

**Issue by jasonkarns** [super-linter/super-linter#5650]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Current log output is not the greatest for digging into violations on github actions.

### Expected Behavior

Now that github actions has [Job Summaries](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/), it would be awesome if super-linter's log output could be redirected directly to the job summary. That way the linter output could be easily consumable from the job page instead of diving into the logs.

**Comment by jasonkarns** [comment]
In the (present) absence of an official feature, is there any existing way to redirect the log output to the job summaries file descriptor now?

I've tried:

```yml
      - uses: github/super-linter/slim@v6
        env:
          # /snip
          CREATE_LOG_FILE: true
          LOG_FILE: ${{ github.step_summary }}
```

but that fails:

```
mv: cannot move '/tmp/tmp.zGd3tC9yAF' to '/github/workspace//github/file_commands/step_summary_a2525f8f-5bd7-40c1-9437-d3d36f44bdf2': No such file or directory
```

**Comment by jasonkarns** [comment]
I've also tried just catting the log file to the summary descriptor but that failed.

```
Run cat super-linter.log >> $GITHUB_STEP_SUMMARY
  cat super-linter.log >> $GITHUB_STEP_SUMMARY
  shell: /usr/bin/bash -e {0}
cat: super-linter.log: Permission denied
Error: Process completed with exit code 1.```

**Comment by echoix** [comment]
I'm not sure that what you'd want as a final solution would work with the current limitations of job summaries.
There is a size limit of the output that can be added to that file. With some linters (I have xmllint in mind), the output in one of my projects was so long that even with the log viewer my browser (on desktop) had a really hard time getting through. The job summary should thus be for a job _summary_, not the complete logs or linter outputs.

For your other issue, you must also keep in mind that the way to write to the job summary is to write to a file that has a path given by an env variable. That file changes for every step, and also had its limits, so a bad step won't break markdown formatting for the subsequent steps that did it correctly. If super-linter is using a composite action type, I don't know how the definition of step is applied (is the whole action a same step, or each step in the composite action given a GITHUB_STEP_SUMMARY?). If it is a docker type, I would expect that in addition to having the  GITHUB_STEP_SUMMARY variable available, that temp folder must be also accessible from inside docker (does GitHub's action runner do it for us for any container launched with the docker action type?). Only after that, I finally would suspect a permission error, as concluded in the issue.


In conclusion, this suggestion might need to be refined a bit to be realistic. And to not duplicate what the CI logs are already designed for. And to keep in mind what the job summary would look like when super-linter is only one of the steps in a workflow's job.

**Comment by jasonkarns** [comment]
> The job summary should thus be for a job summary, not the complete logs or linter outputs.
